edg/parts/Lcd_Er_Tft1_28_3.py

```diff
@@ -0,0 +1,105 @@
+        )))
+        self.gnd = self.Export(self.conn.pins.request('1').adapt_to(Ground()))
+        # Backlight control
+        self.ledk = self.Export(self.conn.pins.request('2'))
+        self.leda = self.Export(self.conn.pins.request('3'))
+
+        dio_model = DigitalBidir.from_supply(
+            self.gnd, self.vdd,
+            voltage_limit_abs=(-0.3 * Volt, self.vdd.voltage_limits.upper()),
+            input_threshold_factor=(0.3, 0.7)
+        )
+
+        self.rs = self.Export(self.conn.pins.request('5').adapt_to(DigitalSink()))
+
+        # Control pins
+        self.spi = self.Port(SpiPeripheral.empty())
+        self.cs = self.Export(self.conn.pins.request('6').adapt_to(dio_model))
+        self.connect(self.spi.sck, self.conn.pins.request('7').adapt_to(dio_model))
+        self.connect(self.spi.mosi, self.conn.pins.request('8').adapt_to(dio_model))
+
+        self.miso_nc = self.Block(DigitalBidirNotConnected())
+        self.connect(self.spi.miso, self.miso_nc.port)
+
+        self.rst = self.Export(self.conn.pins.request('9').adapt_to(DigitalSink()))
+
+        # Capacitive Touch Panel (CTP)
+        self.ctp_i2c = self.Port(I2cTarget(DigitalBidir.empty(), addresses=[0x15]),)
+
+        self.ctp_vdd = self.Export(self.conn.pins.request('10').adapt_to(VoltageSink(
+            voltage_limits=(2.7, 3.6) * Volt,
+            current_draw=(5 * uAmp, 2.5 * mAmp)
+        )))
+
+        self.connect(self.gnd, self.conn.pins.request('11').adapt_to(Ground()))
+
+        self.ctp_rst = self.Export(self.conn.pins.request('12').adapt_to(dio_model))
+        self.ctp_int = self.Export(self.conn.pins.request('13').adapt_to(dio_model))
+        self.connect(self.ctp_i2c.sda, self.conn.pins.request('14').adapt_to(dio_model))
+        self.connect(self.ctp_i2c.scl, self.conn.pins.request('15').adapt_to(dio_model))
+
+
+class Er_Tft_128_3(Lcd, Resettable, Block):
+    """GC9A01-based 1.28" 240x240 TFT, with optional capacitive touch panel."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.ic = self.Block(Er_Tft_128_3_Device())
+        self.gnd = self.Export(self.ic.gnd, [Common])
+        self.pwr = self.Export(self.ic.vdd, [Power])
+        self.spi = self.Export(self.ic.spi)
+        self.cs = self.Export(self.ic.cs)
+        self.dc = self.Export(self.ic.rs)
+        # touch interface
+        self.ctp_i2c = self.Export(self.ic.ctp_i2c, optional=True, doc='Touch panel interface i2c')
+        self.ctp_rst = self.Export(self.ic.ctp_rst, optional=True, doc='Touch panel interface')
+        self.ctp_int = self.Export(self.ic.ctp_int, optional=True, doc='Touch panel interface')
+
+    def contents(self):
+        super().contents()
+        self.connect(self.reset, self.ic.rst)
+        self.require(self.reset.is_connected())
+
+        # self.lcd = self.Block(Er_Tft_128_3_Outline())  # for ic outline
+
+        self.connect(self.ic.ledk.adapt_to(Ground()), self.gnd)
+        forward_current = (1, 30)*mAmp
```

Comment:
```suggestion
        forward_current = (24, 30)*mAmp
```
I would have the current tolerance tighter so the brightness is more reasonable, this still gives a ~10% range